src/Tools/NodeDocumentationMarkdownGenerator/AssemblyHandler.cs

```diff
@@ -0,0 +1,387 @@
+                    var externalLib = t.ClassNode.ExternLibName;
+
+                    // For some reason mscorelib sometimes gets pass to here, so filtering it away.
+                    if (t.CLRType.Assembly.GetName().Name == objectAsmName ||
+                        t.ClassNode.ClassAttributes.HiddenInLibrary) continue;
+
+                    var ctorNodes = t.ClassNode.Procedures
+                        .OfType<ConstructorDefinitionNode>()
+                        .Where(c => c.Access == ProtoCore.CompilerDefinitions.AccessModifier.Public &&
+                                    c.MethodAttributes != null ?
+                                        !c.MethodAttributes.IsObsolete && !c.MethodAttributes.HiddenInLibrary :
+                                        true)
+                        .Cast<AssociativeNode>();
+
+                    var functionNodes = t.ClassNode.Procedures
+                        .OfType<FunctionDefinitionNode>()
+                        .Where(f => !f.MethodAttributes.HiddenInLibrary &&
+                                    !f.MethodAttributes.IsObsolete &&
+                                    f.Name != "_Dispose")
+                        .Cast<AssociativeNode>();
+
+                    var associativeNodes = ctorNodes.Union(functionNodes);
+
+                    foreach (var node in associativeNodes)
+                    {
+                        if (TryGetFunctionDescriptor(node, asm, asm.Location, className, customizationFile, out ReflectionFunctionDescriptor des))
+                        {
+                            descriptors.Add(des);
+                        }
+                    }
+                }
+                catch (Exception)
+                {
+                    continue;
+                }
+            }
+
+            return descriptors;
+        }
+
+        private static List<FunctionDescriptor> GetFunctionDescriptorsFromDs(PathManager pathManager, string dsFilePath)
+        {
+            var descriptors = new List<FunctionDescriptor>();
+
+            var t = new ImportModuleHandler(new ProtoCore.Core(new ProtoCore.Options()));
+            var dsCodeNode = t.Import(dsFilePath, "", "");
+            var classNode = dsCodeNode.CodeNode.Body.OfType<ClassDeclNode>().FirstOrDefault();
+            var associativeNodes = classNode.Procedures;
+            var customizationFile = LibraryCustomizationServices.GetForAssembly(dsFilePath, pathManager);
+            foreach (var node in associativeNodes)
+            {
+                if (TryGetFunctionDescriptor(node, null, dsFilePath, classNode.ClassName, customizationFile, out ReflectionFunctionDescriptor des))
+                {
+                    descriptors.Add(des);
+                }
+            }
+            return descriptors;
+        }
+
+        private static void AddZeroTouchNodesToSearch(IEnumerable<FunctionGroup> functionGroups, ref NodeSearchModel nodeSearchModel)
```

Comment:
I guess you need to call this because you depend on ZT nodes being in search so you can scan them later, can you add a comment?